src/Orckestra.Composer.Cart/Services/Order/OrderHistoryViewService.cs

```diff
@@ -79,23 +79,28 @@ public virtual async Task<OrderHistoryViewModel> GetOrderHistoryViewModelAsync(G
             if (param.CustomerId == null) { throw new ArgumentException(GetMessageOfNull(nameof(param.CustomerId)), nameof(param)); }
             if (string.IsNullOrWhiteSpace(param.Scope)) { throw new ArgumentException(GetMessageOfNullWhiteSpace(nameof(param.Scope)), nameof(param)); }
 
+            var orderQueryResult = await OrderRepository.GetCustomerOrdersAsync(param).ConfigureAwait(false);
+            if (orderQueryResult == null)
+            {
+                return new OrderHistoryViewModel();
+            }
+
             var orderDetailBaseUrl = OrderUrlProvider.GetOrderDetailsBaseUrl(param.CultureInfo);
             var orderStatuses = await LookupService.GetLookupDisplayNamesAsync(new GetLookupDisplayNamesParam
             {
                 CultureInfo = param.CultureInfo,
                 LookupType = LookupType.Order,
                 LookupName = "OrderStatus",
             }).ConfigureAwait(false);
-
-            var orderQueryResult = await OrderRepository.GetCustomerOrdersAsync(param).ConfigureAwait(false);
-
+            
             var shipmentsTrackingInfos = new Dictionary<Guid, TrackingInfoViewModel>();
             var orderSettings = await GetOrderSettings(param.Scope).ConfigureAwait(false);
-            var ordersDetails = await GetOrders(orderQueryResult, param).ConfigureAwait(false);
+            var ordersDetails = new List<Overture.ServiceModel.Orders.Order>();
 
             if (orderQueryResult != null && orderQueryResult.Results != null && param.OrderTense == OrderTense.CurrentOrders)
             {
                 shipmentsTrackingInfos = GetShipmentsTrackingInfoViewModels(ordersDetails, param);
+                ordersDetails = await GetOrders(orderQueryResult, param).ConfigureAwait(false);
```

Comment:
you need load order details before you use it in GetShipmentsTrackingInfoViewModels